components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/NotificationTemplateManagerImpl.java

```diff
@@ -0,0 +1,720 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.email.mgt.constants.TemplateMgtConstants.DEFAULT_EMAIL_NOTIFICATION_LOCALE;
+import static org.wso2.carbon.email.mgt.constants.TemplateMgtConstants.DEFAULT_SMS_NOTIFICATION_LOCALE;
+import static org.wso2.carbon.email.mgt.util.I18nEmailUtil.normalizeLocaleFormat;
+
+/**
+ * Provides functionality to manage notification templates.
+ */
+public class NotificationTemplateManagerImpl implements NotificationTemplateManager {
+
+    private static final Log log = LogFactory.getLog(NotificationTemplateManagerImpl.class);
+
+    private final TemplatePersistenceManager userDefinedTemplatePersistenceManager;
+    private final TemplatePersistenceManager systemTemplatePersistenceManager;
+    private final TemplatePersistenceManager unifiedTemplatePersistenceManager;
+
+    public NotificationTemplateManagerImpl() {
+
+        TemplatePersistenceManagerFactory templatePersistenceManagerFactory = new TemplatePersistenceManagerFactory();
+        this.userDefinedTemplatePersistenceManager =
+                templatePersistenceManagerFactory.getUserDefinedTemplatePersistenceManager();
+        this.systemTemplatePersistenceManager = templatePersistenceManagerFactory.getSystemTemplatePersistenceManager();
+        this.unifiedTemplatePersistenceManager = templatePersistenceManagerFactory.getTemplatePersistenceManager();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+     public List<String> getAllNotificationTemplateTypes(String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerException {
+
+         try {
+             if (OrganizationManagementUtil.isOrganization(tenantDomain)) {
+                 // Return the root organization's template types.
+                 tenantDomain = getRootOrgTenantDomain(tenantDomain);
+             }
+             List<String> templateTypes = unifiedTemplatePersistenceManager
+                     .listNotificationTemplateTypes(notificationChannel, tenantDomain);
+             return templateTypes != null ? templateTypes : new ArrayList<>();
+         } catch (NotificationTemplateManagerServerException ex) {
+             String errorMsg = String.format("Error when retrieving %s template types of %s tenant.",
+                     notificationChannel.toLowerCase(), tenantDomain);
+             throw new NotificationTemplateManagerServerException(errorMsg, ex);
+         } catch (OrganizationManagementException e) {
+             throw new NotificationTemplateManagerServerException(e.getMessage(), e);
+         }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void addNotificationTemplateType(String notificationChannel, String displayName, String tenantDomain)
+            throws NotificationTemplateManagerException {
+
+        validateDisplayNameOfTemplateType(displayName);
+        try {
```

Comment:
At adding , validate to not include any leading or following empty spaces.

Reply:
addressed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/250/commits/4a4dae818d530081e8a91a33d6d70da13c375c9c